dataikuapi/dss/ml.py

```diff
@@ -692,25 +697,282 @@ def compute_partial_dependencies(self, features, wait=True, sample_size=1000, ra
+        self.modalities = [DSSSubpopulationModality(self.computed_as_type, m) for m in self.get("modalities", [])]
+
+    def get_computation_params(self):
+        """
+        Gets computation params
+        """
+        return {
+            "nbRecords":  self.get("nbRecords"),
+            "randomState":  self.get("randomState"),
+            "onSample":  self.get("onSample")
+        }
+    
+    def list_modalities(self):
+        """
+        List definitions of modalities
+        """
+        return [m.definition for m in self.modalities]
+
+    def get_modality(self, definition=None):
+        """
+        Retrieves modality from definition
+
+        :param definition: definition of modality to retrieve. Can be:
+                   * :class:`dataikuapi.dss.ml.DSSSubpopulationModalityDefinition`
+                   * `dataikuapi.dss.ml.DSSSubpopulationModalityDefinition.MISSING_VALUES` 
+                      to retrieve modality corresponding to missing values
+                   * for category modality, can be a str corresponding to the value of the modality
+                   * for numeric modality, can be a number inside the modality
+
+        :returns: the modality
+        :rtype: :class:`dataikuapi.dss.ml.DSSSubpopulationModality`
+        """
+
+        if definition == DSSSubpopulationModalityDefinition.MISSING_VALUES:
+            for m in self.modalities:
+                if m.definition.missing_values:
+                    return m
+            raise ValueError("No 'missing values' modality found")
+
+        if isinstance(definition, DSSSubpopulationModalityDefinition):
+            modality_candidates = [m for m in self.modalities if m.definition.index == definition.index]
+            if len(modality_candidates) == 0:
+                raise ValueError("Modality with index '%s' not found" % modality["index"])
+            return modality_candidates[0]
+        
+        for m in self.modalities:
+            if m.definition.contains(definition):
+                return m
+        raise ValueError("Modality not found: %s" % definition)
+
+    def get_raw(self):
+        """
+        Gets the raw dictionary of the subpopulation analysis
+        """
+        return self.internal_dict
+
+
+class DSSSubpopulationAnalyses(DSSExtensibleDict):
+    """
+    Object to read details of subpopulation analyses of a trained model
+
+    Do not create this object directly, use :meth:`DSSTrainedPredictionModelDetails.get_subpopulation_analyses()` instead
+    """
+
+    def __init__(self, data):
+        super(DSSSubpopulationAnalyses, self).__init__(data)
+        self.analyses = []
+        for analysis in data.get("subpopulationAnalyses", []):
+            self.analyses.append(DSSSubpopulationAnalysis(analysis))
+    
+    def get_raw(self):
+        """
+        Gets the raw dictionary of subpopulation analyses
+        """
+        return self.internal_dict
+    
+    def get_all_dataset(self):
```

Comment:
Not the usual "dataset", and may make it seem like it's the full unsampled dataset. Rename:

```suggestion
    def get_global(self):
```

Reply:
In that case, we should probably rename the field in the data structure as well, it's decided in memory, so no need for migration

Reply:
Sure